### PR TITLE
Update IPAM documentation.

### PR DESCRIPTION
**Pull request by marccarre**
- Added section about persistence.
- Updated section about claiming an address.

### CODE REVIEW

File: docs/ipam.md [hunk 1]
```diff
@@ -65,14 +65,30 @@ peer for space:
...
 
+### Data persistence
+
+Key IPAM data is saved to disk, in a [BoltDB](https://github.com/boltdb/bolt)
+file, stored within a [data volume container](https://docs.docker.com/engine/userguide/containers/dockervolumes/#creating-and-mounting-a-data-volume-container)
+named `weavedb`.
+
+This file contains:
+
+* the division of the IP allocation range amongst peers, and
```
Comment (brb) [inline docs/ipam.md]: Maybe it's worth mentioning that we persist more data (e.g. a local peer name), and the listed one is relevant for IPAM.

Comment (bboreham) [inline docs/ipam.md]: suggest "This file contains" -> "This file is used to persist data for various Weave Net components; for IPAM it contains:"

Comment (marccarre) [inline docs/ipam.md]: thanks for the suggestions -- updated accordingly.

File: docs/ipam.md [hunk 2]
```diff
@@ -65,14 +65,30 @@ peer for space:
...
-individual IP addresses are assigned to containers in order to avoid
-giving the same address out in subsequent allocation requests. The
-weave script invokes the `claim` command in `populate_ipam` to do
-this.
+If a Weave process is restarted,
+
+- if `weavedb` is present, then it loads persisted IPAM data from there, as
+described, in the previous section;
+- else, it learns from other peers which ranges it used to own.
+The `weave` script then invokes the `claim` command via the `ipam_reclaim`
```
Comment (brb) [inline docs/ipam.md]: As per https://github.com/weaveworks/weave/pull/2787, `ipam_reclaim` no longer exists.

Comment (marccarre) [inline docs/ipam.md]: I've removed these details from the documentation completely.
Such technical details are likely to get out of date faster than the overall design/flow.

**Comment by brb** [comment]
LGTM.

**Pull request merged**